### ISSUE TITLE
deepspeed  must?

### ISSUE THREAD

**Issue by ex-yanminmin001** [horus-ai-labs/DistillFlow#11]
i want to train the model by single gpu and my yaml like :
`student_model:
  model_name_or_path: /LLMs/llm/Qwen2.5-1.5B-Instruct/
  flash_attn: fa2
  use_unsloth: false
  output_attentions: false
  # deepspeed_config: ""
  # DeepSpeed: disable
  
  # chat_template: "{% for message in messages %}{% if loop.first and messages[0]['role'] != 'system' %}{{ '<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n' }}{% endif %}{{'<|im_start|>' + message['role'] + '\n' + message['content'] + '<|im_end|>' + '\n'}}{% endfor %}{% if add_generation_prompt %}{{ '<|im_start|>assistant\n' }}{% endif %}"

teacher_model:
  model_name_or_path: /LLMs/llm/Qwen2.5-7B-Instruct/
  flash_attn: fa2
  use_unsloth: false
  output_attentions: false
  # deepspeed_config: ""
  

data:
  seed: 42
  text_field: "text"
  train_datasets:
    - path: /datasets/sft_data/
      template:
        name: alpaca
      num_samples: 100
  test_size: 0.2
  streaming: false

distill:
  type: logits
  max_seq_length: 1024
  resume_from_checkpoint: null
  temperature: 2.0
  alpha: 0.5
  sft_config:
    output_dir: './results'
    num_train_epochs: 1
    per_device_train_batch_size: 1
    gradient_accumulation_steps: 8
    eval_strategy: steps
    eval_steps: 100
    save_steps: 2000
    logging_steps: 1
    learning_rate: 2.0e-5
    weight_decay: 0.05
    warmup_ratio: 0.1
    lr_scheduler_type: 'cosine'
    fp16: False
    bf16: True
    max_grad_norm: 1.0
    group_by_length: False
    remove_unused_columns: false
`
and my start cmd like 
`CUDA_VISIBLE_DEVICES=5 python src/trainer.py --config local_distill.yaml
`
i got the error

![Image](https://github.com/user-attachments/assets/7706c5a7-0a96-4625-940b-d80aab3781f4)
can you help me?thanks a lot!

**Comment by karan15234** [comment]
@ex-yanminmin001 you caan use deepspeed zero config to disable deepspeed. Will that work for you?
```
deepspeed_config: './deepspeed/zero0.json'
```

**Comment by horus-labs-ml** [comment]
Please use the command to launch your training. 
accelerate launch src/trainer.py --config <your_config_path>

**Comment by ex-yanminmin001** [comment]
> Please use this command to launch your training. accelerate launch src/trainer.py --config <your_config_path>

same error
